src/Catalyst.Core.Modules.Dfs/BlockExchange/BitSwapService.cs

```diff
@@ -0,0 +1,557 @@
+        private readonly ConcurrentDictionary<Peer, BitswapLedger> _peerLedgers = new ConcurrentDictionary<Peer, BitswapLedger>();
+
+        /// <summary>
+        ///   The supported bitswap protocols.
+        /// </summary>
+        /// <value>
+        ///   Defaults to <see cref="Bitswap11"/> and <see>
+        ///       <cref>Bitswap1</cref>
+        ///   </see>
+        ///   .
+        /// </value>
+        public IBitswapProtocol[] Protocols { get; set; }
+
+        /// <summary>
+        ///   The number of blocks sent by other peers.
+        /// </summary>
+        private ulong _blocksReceived;
+
+        /// <summary>
+        ///   The number of bytes sent by other peers.
+        /// </summary>
+        private ulong _dataReceived;
+
+        /// <summary>
+        ///   The number of blocks sent to other peers.
+        /// </summary>
+        private ulong _blocksSent;
+
+        /// <summary>
+        ///   The number of bytes sent to other peers.
+        /// </summary>
+        private ulong _dataSent;
+
+        /// <summary>
+        ///   The number of duplicate blocks sent by other peers.
+        /// </summary>
+        /// <remarks>
+        ///   A duplicate block is a block that is already stored in the
+        ///   local repository.
+        /// </remarks>
+        private ulong _dupBlksReceived;
+
+        /// <summary>
+        ///   The number of duplicate bytes sent by other peers.
+        /// </summary>
+        /// <remarks>
+        ///   A duplicate block is a block that is already stored in the
+        ///   local repository.
+        /// </remarks>
+        private ulong _dupDataReceived;
+
+        /// <summary>
+        ///   Creates a new instance of the <see cref="BitSwapService"/> class.
+        /// </summary>
+        public BitSwapService()
+        {
+            Protocols = new IBitswapProtocol[]
+            {
+                new Bitswap11 {BitswapService = this}
+            };
+        }
+
+        /// <summary>
+        ///   Provides access to other peers.
+        /// </summary>
+        public SwarmService SwarmService { get; set; }
+
+        /// <summary>
+        ///   Provides access to blocks of data.
+        /// </summary>
+        public IBlockApi BlockService { get; set; }
+
+        /// <summary>
+        ///   Statistics on the bitswap component.
+        /// </summary>
+        /// <seealso cref="IStatsApi"/>
+        public BitswapData Statistics
+        {
+            get
+            {
+                return new BitswapData
+                {
+                    BlocksReceived = _blocksReceived,
+                    BlocksSent = _blocksSent,
+                    DataReceived = _dataReceived,
+                    DataSent = _dataSent,
+                    DupBlksReceived = _dupBlksReceived,
+                    DupDataReceived = _dupDataReceived,
+                    ProvideBufLen = 0, // TODO: Unknown meaning
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Complete the task associated to this 'TODO' comment.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)